_posts/2025-08-16-accessible-error-messages-in-forms.md

```diff
@@ -0,0 +1,158 @@
+HTML5 form validation is not accessible at this moment. The W3C summarizes the options in [Validating Input ](https://www.w3.org/WAI/tutorials/forms/validation/). "If your web browser supports HTML5, it will not allow you to submit the form without entering text into the input field. Instead, it will display a message that is generated by the web browser itself".  Write custom error messages and provide server side generated error messages. The W3C explains the options in [Validating Input ](https://www.w3.org/WAI/tutorials/forms/validation/).
+
+## How do I indicate that an answer is incomplete or filled out incorrectly?
+
+Use more than colour alone to indicate errors. A user with visual impairments or colour blindness may not notice a red outline.  Colour is helpful — but always include an error message in text too.
+
+* Clearly state whether a field is required.
+* Show errors with more than just colour or an icon.
+* Write out the error in text above the form field.
+* Add a summary of all errors above the form  helps the user quickly see what needs to be corrected.
+
+
+## What makes a good error message?
+
+Messages like “This field is required” or “Invalid value” don’t offer much help. Write clear error messages that explain what’s missing or needs changing.
+
+{: .example }
+“Enter your first name”  
+“Your chosen password is too short. It must be at least 8 characters long.”  
+“The expiration date must be in the future.”
+
+This is far more helpful than a vague generic message.
+
+
+## How do I confirm a form was successfully submitted?
+
+You click "Send" and… nothing happens. Or you're redirected to the homepage. Was the form actually sent? How to know?
+
+Give users clear confirmation that their form has been submitted and what will happen next. You can also include this messsage in a confirmation email.
+
+{: .example }
+A confirmation text could be:  
+Thank you for your registration for our workshop "Knitting socks". A confirmation email has been sent to user@example.com with the time and location of the workshop. If you haven't received an email? [Please contact us](#).
+
+## How can I offer help if a user sets stuck?
+
+If a user gets stuck filling out a complex form, they should be able to easily request help.
+
+Don’t hide contact details at the very bottom of the form — place a short sentence near the top. The footer is also a good place to additionally include contact info. Always offer multiple ways to get in touch on your contact page — not everyone can make a phone call.
+
+{: .example }
+A help text could be:  
+Do you need help filling out this form or do you have questions? [Please contact us](#).
+
+## Summary
+
+Help your users as much as possible when filling out a form. Don’t ask unnecessary questions, just for your statistics. Provide the right input options. Make it clear whis is requited and how to fill out a field. Use clear, timely error messages. And make sure it’s easy to ask for help when needed.
```

Comment:
"Don't ask unnecessary questions just to collect for your statistics."

Instead of "Provide the right input options", "Use standard input form fields wherever possible."

Typo: "whis" => "what"
Typo: "requited" => "required"